File: backend/app/schemas/scadenza.py
from datetime import date, datetime
from pydantic import BaseModel, ConfigDict


class ScadenzaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    documento_id: int | None = None
    contratto_id: int | None = None
    cliente_id: int
    tipo_scadenza: str = "contratto"
    descrizione: str | None = None
    data_inizio: date | None = None
    data_scadenza: date | None = None
    durata: str | None = None
    rinnovo_automatico: bool | None = None
    preavviso_disdetta: str | None = None
    canone: str | None = None
    parti_coinvolte: list[str] | None = None
    clausole_chiave: list[str] | None = None
    confidence_score: float
    verificato: bool
    created_at: datetime
    updated_at: datetime


class ScadenzaListOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    documento_id: int | None = None
    contratto_id: int | None = None
    cliente_id: int
    cliente_nome: str
    file_name: str = "Contratto manuale"
    tipo_scadenza: str
    descrizione: str | None = None
    data_scadenza: date | None
    data_inizio: date | None = None
    giorni_rimanenti: int | None = None
    canone: str | None = None
    rinnovo_automatico: bool | None = None
    preavviso_disdetta: str | None = None
    confidence_score: float
    verificato: bool
    is_contratto: bool = False
    created_at: datetime

File: frontend/src/types/scadenza.ts
export interface Scadenza {
    id: number;
    documento_id: number | null;
    contratto_id: number | null;
    cliente_id: number;
    cliente_nome: string;
    file_name: string;
    tipo_scadenza: string;
    descrizione: string | null;
    data_scadenza: string | null;
    data_inizio: string | null;
    giorni_rimanenti: number | null;
    canone: string | null;
    rinnovo_automatico: boolean | null;
    preavviso_disdetta: string | null;
    confidence_score: number;
    verificato: boolean;
    is_contratto: boolean;
    created_at: string;
}

export interface ScadenzaFilters {
    tipo_scadenza?: string;
    cliente_id?: number;
    da_data?: string;
    a_data?: string;
    verificato?: boolean;
    search?: string;
}
